### DISCUSSION TITLE
Multiselect issue

### DISCUSSION THREAD

**Discussion by c-gross** [orchidsoftware/platform#2195] [Q&A] [answered]
Hi, using selects is working fine but when I use select with the multiple method, it's saving the values but it's not showing the selected values when I go to the edit page again.

The db field is json and I can see the json string in the db field.

```
Select::make('product.tags')
                ->title(__('Tags'))
                ->multiple()
                ->options([
                    1  => 'Tag 1',
                    2  => 'Tag 2',
                ]),
```

**Comment by rowanfuchs** [comment, marked as answer]
there is a tiny bug I would say, which is that the multiple selects values which I also stored as a json array into the database are stored as strings. This resulted the in_array function within the blade to fail. What I did was creating a custom select which only uses my custom view with the bug fix that casts the key to be a string.

```php
<?php

declare(strict_types=1);

namespace App\Orchid\Fields;

use Orchid\Screen\Fields\Select;

/**
 * Class Custom Select.
 **/
class CustomSelect extends Select
{
    /**
     * @var string
     */
    protected $view = 'fields.custom.select';

}
```

note the string type cast
```php
@component($typeForm, get_defined_vars())
    <div data-controller="select">
        <select {{ $attributes }}>
            @foreach($options as $key => $option)
                <option value="{{$key}}"
                        @isset($value)
                        @if (is_array($value) && in_array((string) $key, $value, true)) selected
                        @elseif (isset($value[$key]) && $value[$key] == $option) selected
                        @elseif ($key === $value) selected
                    @endif
                    @endisset
                >{{$option}}</option>
            @endforeach
        </select>
    </div>
@endcomponent
```

**Reply by c-gross** [reply]
Had a week of but was finally able to test it and it works fine.
Thank you.

Maybe you should do an PR with that fix?

**Reply by rowanfuchs** [reply]
I've thought about it, although it helps us resolve the issue I feel like the solution should be done in the core. Sadly I'm not sure where.

**Reply by c-gross** [reply]
Yes in vendor/orchid/platform/resources/views/fields/select.blade.php
Guess it's easier to modify on Github and send as change request (https://github.com/orchidsoftware/platform/edit/master/resources/views/fields/select.blade.php)

**Reply by rowanfuchs** [reply]
Yea no I mean within the core functions and not within a blade as some choose a text column where as we casted arrays into json
